ij/src/main/kotlin/org/jetbrains/academy/test/system/ij/analyzer/IjCodeAnalyzerUtil.kt

```diff
@@ -79,3 +80,55 @@ fun PsiFile.findMethodsWithContent(content: String): List<String> =
         val methods = extractElementsOfTypes(KtNamedFunction::class.java)
         methods.filter { it.getBlockBody() == formattingContent }.mapNotNull { it.name }.toList()
     }
+
+/**
+ * Checks if the PsiFile contains a property with the specified property name.
+ *
+ * @param propertyName The name of the property to search for.
+ * @return True if the PsiFile contains a property with the given name, false otherwise.
+ */
+fun PsiFile.hasProperty(propertyName: String): Boolean = ApplicationManager.getApplication().runReadAction<Boolean> {
+    extractElementsOfTypes(KtProperty::class.java).any { it.name == propertyName }
+}
+
+/**
+ * Checks if the PsiFile contains a method with the specified method name.
+ *
+ * @param methodName The name of the method to search for.
+ * @return True if the PsiFile contains a method with the given name, false otherwise.
+ */
+fun PsiFile.hasMethod(methodName: String): Boolean = ApplicationManager.getApplication().runReadAction<Boolean> {
+    extractElementsOfTypes(KtNamedFunction::class.java).any { it.name == methodName }
+}
+
+/**
+ * Retrieves the text of the parent element of the given PsiElement.
+ *
+ * @param element The PsiElement for which to find the parent.
+ * @param isParentTypeFunction If true, the parent element is expected to be a function (KtNamedFunction).
+ *                            If false, the parent element is expected to be a class or any other non-function type.
+ * @return The text of the parent element if found, or "no name" if the parent is not found or has no name (for functions).
+ */
+private fun getParent(element: PsiElement, isParentTypeFunction: Boolean): String {
+    return if (isParentTypeFunction) {
+        element.parentsOfType(KtNamedFunction::class.java).first().name ?: "no name"
+    } else {
+        element.parent.text
```

Comment:
According to the documentation, `parent` could be null, so let's use `?.` here. Just in case :-)